src/manageiq_client/api.py

```diff
@@ -616,11 +616,33 @@ def __call__(self, *args, **kwargs):
             self.api.response = action_response
         return outcome
 
+    def _get_entity_from_href(self, result):
+        href_result = result['href']
+        href_match = None
+
+        # If the "href" value in result doesn't match the current collection,
+        # try to find the collection that the "href" refers to.
+        if self.collection._href not in href_result:
+            href_match = re.match(r"(http[^?]+)/([a-z_-]+)", href_result)
+
+        if href_match:
+            collection_name = href_match.group(2)
+            entry_point = href_match.group(1)
+            if collection_name and entry_point:
+                new_collection = Collection(
+                    self.collection.api,
+                    "{}/{}".format(entry_point, collection_name),
+                    collection_name
+                )
+                return Entity(new_collection, result, incomplete=True)
+
+        return Entity(self.collection, result, incomplete=True)
```

Comment:
The intent was to be backwards compatible and fall through if the attempt to find "better" collection failed. Even in wrong collection the data can still be usable, because the href points to an existing location (it can be reloaded, get up-to-date message etc.).